staging/script.xsqueeze/resources/lib/classes/NowPlayingWindow.py

```diff
@@ -295,7 +302,41 @@ def update(self):
         self.updatePlaylistDetails()
         self.updateCoverArtFromURLs()
 
-  ##############################################################################
+        #trigger a hdmi command, if a controller has changed the power status or the volume
+        if constants.TURNONAVR == 'true' and constants.CECSUPPORT == 'true':
+          powerstate = self.player.getpowerstate()
+          if powerstate == 0 and constants.HDMIPOWER == "true":
+            if constants.TURNOFFTV == "true":
+               tv.standby()
+               pass
+            if constants.TURNONAVR == "true":
+               avr.standby()
+            constants.HDMIPOWER = "false"
+          elif powerstate != 0 and constants.HDMIPOWER == "false":
+            if constants.TURNONAVR == "true":
+               avr.power_on()
+               avr.set_av_input(int(constants.AVRINPUT))
+            constants.HDMIPOWER = "true"
+          else:
+            pass
+
+        if constants.ADJUSTVOL == "true" and constants.CECSUPPORT == 'true':
+          vol1 = self.player.getVolume()
+          xbmc.sleep(1000)
```

Comment:
Why two calls to get volume, and why a sleep?  You're in a loop here, that operates basically every half a second anyway.  It then grabs a thread lock and should process everything quickly before sleeping the half second.